CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java

```diff
@@ -35,25 +38,16 @@ public void initialize() {
     }
 
     @Override
-    public void execute() {
-        BallSpatialInfo[] ballInfo = visionSubsystem.getBoulderInfo();
-        
-        if(!visionSubsystem.isConnectionHealthy() || ballInfo == null || ballInfo.length <= 0) {
+    public void execute() {        
+        if(!visionSubsystem.isConnectionHealthy() || !visionSubsystem.trackingAnyBalls()) {
            driveSubsystem.stopDrive();
         }
         else {
-            BallSpatialInfo targetBall = null;
-            for(BallSpatialInfo ball : ballInfo) {
-                if(targetBall == null || ball.distanceInches > targetBall.distanceInches) {
-                    targetBall = ball;
-                }
-            }
+            BallSpatialInfo targetBall = visionSubsystem.findTargetBall();
 
-            double newRotationalPower = rotationalPidManager.calculate(0, targetBall.relativeHeading);
+            double newRotationalPower = rotationalPidManager.calculate(cameraCenterHeading.get(), targetBall.relativeHeading);
```

Comment:
Is there a potential race condition where targetBall could be null due to the subsystem updating on an independent thread?